modules/rtpproxy/rtpproxy.c

```diff
@@ -2120,9 +2148,16 @@ send_rtpp_command(struct rtpp_node *node, struct iovec *v, int vcnt)
 				break;
 			}
 		}
-		v[0].iov_base = gencookie();
-		v[0].iov_len = strlen(v[0].iov_base);
-		for (i = 0; i < rtpproxy_retr; i++) {
+		if (!CM_STREAM(node)) {
+			v[0].iov_base = gencookie();
+			v[0].iov_len = strlen(v[0].iov_base);
+		} else {
+			memmove(v, v + 1, (vcnt - 1) * sizeof(*v));
```

Comment:
I am pretty sure that modifying `v` is not correct, as if the current node fails, the function above (`force_rtp_proxy_body`) might failover to a new node, that runs over UDP, so it will need to store the cookie in `v[0]`.
It would probably be easier if we always add the last `\n`, and if the node is stream based, use `v+1`, otherwise `v`.

Reply:
Good catch, Razvan, thanks for review. I'll work on a fix. I did not like that memmove(), anyhow. 👍 :)